@here/harp-mapview/lib/PickingRaycaster.ts

```diff
@@ -5,21 +5,22 @@
  */
 
 import * as THREE from "three";
-import { MapView } from "./MapView";
 
 /**
  * Raycasting points is not supported as necessary in Three.js. This class extends a
- * [[THREE.Raycaster]] except that it holds a reference to [[MapView]] allowing to access the
- * camera and the renderer from [[MapViewPoints]], in order to project the points in screen space.
+ * [[THREE.Raycaster]] by holding a reference to a [[Camera]] allowing to project the points in
+ * screen space, used for example in the [[MapViewPoints]].
  */
 export class PickingRaycaster extends THREE.Raycaster {
     /**
-     * Constructs a `MapViewRaycaster`. It keeps a reference to [[MapView]] in order to access the
-     * active camera when calling the custom method `raycastPoints`.
+     * Constructor.
      *
-     * @param m_mapView the active [[MapView]].
+     * @param camera the camera used for projection to screen. Note, this must use the camera used
+     * to finally render the scene and not the camera in world space.
+     * @param width the canvas width.
+     * @param height the canvas height.
      */
-    constructor(public mapView: MapView) {
+    constructor(public camera: THREE.Camera, public width: number, public height: number) {
```

Comment:
Why do we still need the camera as a member? You should just forward the camera to the Raycaster constructor